src/PhpWord/Reader/Word2007/AbstractPart.php

```diff
@@ -160,6 +160,15 @@ protected function readParagraph(XMLReader $xmlReader, \DOMElement $domNode, $pa
                     $this->readRun($xmlReader, $node, $textContent, $docPart, $paragraphStyle);
                 }
             }
+
+            // Every heading is generaly associated to a bookmark.
+            // Get the name of the current title bookmark and add a similar bookmark before the title
+            $bookmarkTitleNodes = $xmlReader->getElements('w:bookmarkStart', $domNode);
+            if ($bookmarkTitleNodes->length) {
```

Comment:
You should only use booleans in an if condition: `if ($bookmarkTitleNodes->length !== 0) {`

Reply:
There is a problem with the `getElements` method. The DocBlock of `getElements` says it returns a `DOMNodeList` but it actually can also return `false` or an `array`. You should check for that here as well (or let the getElements method throw an exception if it returns false).